### PR TITLE
[bugFix] program import error 

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

* **Issue:** Closes https://app.clickup.com/t/8695xjma3

### :memo: Implementation
- [x] Fix orgunit bug in Bulk load program import
- [x] Get tracker job report only when tracker jobs are complete

### :video_camera: Screenshots/Screen capture

https://github.com/user-attachments/assets/2488942c-8ed9-45c0-9fcc-32c22b9a1ab1

### :bookmark_tabs: Others
#8695xjma3

### CODE REVIEW

**Comment by ifoche** [comment]
Task linked: [CU-8695xjma3 bug in program import between 3.23.2 and 3.24.1](https://app.clickup.com/t/8695xjma3)

File: src/data/Dhis2RelationshipTypes.ts [hunk 1]
```diff
@@ -239,7 +239,7 @@ async function getConstraintForTypeTei(
 
     const ouModeQuery =
         ouMode === "SELECTED" || ouMode === "CHILDREN" || ouMode === "DESCENDANTS"
-            ? { orgUnitMode: ouMode, orgUnit: organisationUnits?.map(({ id }) => id) }
+            ? { orgUnitMode: ouMode, orgUnit: organisationUnits?.map(({ id }) => id).join(";") }
```
Comment (eperedo) [inline src/data/Dhis2RelationshipTypes.ts:242]: This code 
```ts
organisationUnits?.map(({ id }) => id).join(";")
```

is being used in the file `src/data/Dhis2TrackedEntityInstances.ts` as well. Can we create a function `buildOrgUnitParameter` or something similar and reuse it?

```ts
function buildOrgUnitParameter(orgUnitsIds: Maybe<Id[]>): string {
 return orgUnitsIds?.map(({ id }) => id).join(";")
}
```

**Review by eperedo — CHANGES_REQUESTED** [review]
Thanks @deeonwuli. I'm always getting this error about required values not being filled. I using the [program product register](https://dev.eyeseetea.com/who-dev-238/dhis-web-maintenance/index.html#/edit/programSection/program/G6ChA5zMW9n) and this is the file [AMC - Product Register (13).xlsx](https://github.com/user-attachments/files/17157130/AMC.-.Product.Register.13.xlsx)


image

File: src/data/Dhis2RelationshipTypes.ts (hunk 1 above)
Comment (deeonwuli) [reply src/data/Dhis2RelationshipTypes.ts:242]: okay got it!

**Comment by MiquelAdell** [comment]
Hi @deeonwuli is this ready to request a review again from @eperedo ?

Thanks!

**Review by eperedo — APPROVED** [review]
[code review only] Thanks @deeonwuli looks good.

**Review by MiquelAdell — APPROVED** [review]
tested uploading both the file in the clickup task as well as the one that eduardo shared in the github discussion and both imported without issue

**Pull request merged**